File: backend/src/ArchiSubscription.Core/Entities/Customer.cs
namespace ArchiSubscription.Core.Entities;

/// <summary>
/// Represents a bank customer who can have multiple subscriptions.
/// </summary>
public class Customer : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;

    // Navigation property
    public ICollection<Subscription> Subscriptions { get; set; } = new List<Subscription>();
}

File: mobile/app/src/main/java/com/archi/subscription/data/model/Customer.kt
package com.archi.subscription.data.model

data class Customer(
    val id: String,
    val fullName: String,
    val email: String,
    val phoneNumber: String,
    val createdAt: String,
    val activeSubscriptionCount: Int = 0
)

data class CreateCustomerRequest(
    val fullName: String,
    val email: String,
    val phoneNumber: String
)

